packages/cli/src/commands/upgradeHandler.js

```diff
@@ -0,0 +1,240 @@
+import fs from 'fs'
+import path from 'path'
+
+import execa from 'execa'
+import latestVersion from 'latest-version'
+import Listr from 'listr'
+import VerboseRenderer from 'listr-verbose-renderer'
+import terminalLink from 'terminal-link'
+
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths } from '../lib'
+import c from '../lib/colors'
+import { generatePrismaClient } from '../lib/generatePrismaClient'
+
+// Used in yargs builder to coerce tag AND to parse yarn version
+const SEMVER_REGEX =
+  /(?<=^v?|\sv?)(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*)(?:\.(?:0|[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*))*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?(?=$|\s)/i
```

Comment:
I didn't change this at all, just moved it from one file to another.

Reply:
Yeah, this is fine. If someone did figure out string that's super expensive to run this regex against it would still only be possible to run on that user's own machine at dev time. It's not something that can be used to target running servers.